### PR TITLE
Add index generation

### PR DESCRIPTION
**Pull request by joel-lim**
Add support for generating index

### CODE REVIEW

File: enhance_textbook/inputs/include.csv [hunk 1]
```diff
@@ -0,0 +1 @@
...equirements traceability,requirements traceability matrix,review,review tool,reviewer,risk,risk analysis,risk category,risk control,risk identification,risk level,risk management,risk mitigation,risk type,safety,scalability,scenario,scope,scribe,scrum,security,sequence diagram,signoff,software,software failure mode and effect analysis (sfmea),software failure mode, effects, and criticality analysis (sfmeca),software lifecycle,software quality,software quality characteristic,software process improvement,solution,solution model,solution validation,subject matter expert,solution specification,specification,stability,stakeholder,standard,state machine,state transition,state machine diagram,storytelling,structure diagram,suitability,swot,sysml,system,system analysis,system analyst,system boundary,system interface analysis,systems modeling language,technical review,testability,testable requirements,timing diagram,traceability,trend,trial and error,unified modeling language,understandability,usability,use case,use case diagram,user,user innovation,value,validation,vendor,verification,version,version control,vertical traceability,vision,v-model,walkthrough,workshop,work breakdown structure
```
Comment (damithc) [inline enhance_textbook/inputs/include.csv:1]: Would it be better to have one value per line. Easier for manual editing (no need for horizontal scrolling)

**Review by damithc — COMMENTED** [review]
@joel-lim can add some user documentation of this feature somewhere?

File: enhance_textbook/inputs/include.csv (hunk 1 above)
Comment (joel-lim) [reply enhance_textbook/inputs/include.csv]: Alright, I've edited the csv files to be one value per line and added some documentation on the README

**Comment by damithc** [comment]
@joel-lim I'm running into an error at this step:
> 4. `pip install -r requirements.txt`

Can you go through the same steps and confirm they work for you?

**Comment by joel-lim** [comment]
> 
> 
> @joel-lim I'm running into an error at this step:
> 
> > 1. `pip install -r requirements.txt`
> 
> Can you go through the same steps and confirm they work for you?

@damithc It seems to work fine, I was able to install the requirements inside a fresh virtualenv. What's the error message that you received when trying to install the requirements?

Also let me update the outdated repository/folder name in the README!

**Comment by damithc** [comment]
> @damithc It seems to work fine, I was able to install the requirements inside a fresh virtualenv. What's the error message that you received when trying to install the requirements?

Could be a Windows-specific problem.
>Collecting spacy==2.3.2
  Using cached spacy-2.3.2.tar.gz (5.9 MB)
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\repos\nus-se-instructors\pdf-annotation\.env\scripts\python.exe' 'c:\repos\nus-se-instructors\pdf-annotation\.env\lib\site-packages\pip' install --ignore-installed --no-user --prefix '
C:\Users\damithch\AppData\Local\Temp\pip-build-env-8z4v1nfg\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cython>=0.25' 'cy
mem>=2.0.2,<2.1.0' 'preshed>=3.0.2,<3.1.0' 'murmurhash>=0.28.0,<1.1.0' thinc==7.4.1
       cwd: None
  Complete output (97 lines):
  Collecting setuptools
    Using cached setuptools-50.3.0-py3-none-any.whl (785 kB)
  Collecting wheel
    Using cached wheel-0.35.1-py2.py3-none-any.whl (33 kB)
  Collecting cython>=0.25
    Using cached Cython-0.29.21-cp36-cp36m-win32.whl (1.5 MB)
  Processing c:\users\damithch\appdata\local\pip\cache\wheels\7a\31\20\8371d8dd53b773c31f507c6fddcd8bfcb2b76d26bf8334dfc1\cymem-2.0.3-cp36-cp36m-win32.whl
  Processing c:\users\damithch\appdata\local\pip\cache\wheels\46\4a\86\d8e2ccdd36a2484d234e3463f8d6667a0c043c130012c9cbdb\preshed-3.0.2-cp36-cp36m-win32.whl
  Processing c:\users\damithch\appdata\local\pip\cache\wheels\bf\8a\33\62840cd27257a675c9608467daf2cfeb40ece7d7258035e33e\murmurhash-1.0.2-cp36-cp36m-win32.whl
  Processing c:\users\damithch\appdata\local\pip\cache\wheels\f0\64\6d\6499f114a2d6d959d5dbb7c2937b83feeaca6d2c86a2475114\thinc-7.4.1-cp36-cp36m-win32.whl
  Collecting blis<0.5.0,>=0.4.0
    Using cached blis-0.4.1.tar.gz (1.8 MB)
  Collecting wasabi<1.1.0,>=0.0.9
    Using cached wasabi-0.8.0-py3-none-any.whl (23 kB)
  Collecting numpy>=1.7.0
    Using cached numpy-1.19.2-cp36-cp36m-win32.whl (10.9 MB)
  Collecting plac<1.2.0,>=0.9.6
    Using cached plac-1.1.3-py2.py3-none-any.whl (20 kB)
  Collecting tqdm<5.0.0,>=4.10.0
    Using cached tqdm-4.49.0-py2.py3-none-any.whl (69 kB)
  Collecting catalogue<1.1.0,>=0.0.7
    Using cached catalogue-1.0.0-py2.py3-none-any.whl (7.7 kB)
  Processing c:\users\damithch\appdata\local\pip\cache\wheels\50\02\f5\4a2f8d1916fc3d81dfd3ea1223ed81b7939e6f436c52596097\srsly-1.0.2-cp36-cp36m-win32.whl
  Collecting importlib-metadata>=0.20; python_version < "3.8"
    Using cached importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
  Collecting zipp>=0.5
    Using cached zipp-3.1.0-py3-none-any.whl (4.9 kB)
  Building wheels for collected packages: blis
    Building wheel for blis (setup.py): started
    Building wheel for blis (setup.py): finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: 'c:\repos\nus-se-instructors\pdf-annotation\.env\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\damithch\\AppData\\Local\\Temp\\pip-install-ac
2aoxfa\\blis\\setup.py'"'"'; __file__='"'"'C:\\Users\\damithch\\AppData\\Local\\Temp\\pip-install-ac2aoxfa\\blis\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replac
e('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\damithch\AppData\Local\Temp\pip-wheel-p2isqjd0'
         cwd: C:\Users\damithch\AppData\Local\Temp\pip-install-ac2aoxfa\blis\
...

**Comment by damithc** [comment]
There's a lot more in the error message, including these two lines
>ERROR: Failed building wheel for blis
    Running setup.py clean for blis

Probably this problem https://stackoverflow.com/questions/59920225/installing-spacy-with-pip-failed-with-error-failed-building-wheel-for-blis

I'm using 64 bit Python BTW

**Comment by joel-lim** [comment]
Alright let me try to find a fix!

**Comment by joel-lim** [comment]
I decided to remove spacy - I think I was using it quite needlessly (pretty much just for string searching for the index terms) so I replaced it with another library that implements the Aho-Corasick algorithm for fast string-searching.

File: requirements.txt [hunk 2]
```diff
@@ -2,3 +2,5 @@ beautifulsoup4==4.5.3
 PyMuPDF==1.17.5
 selenium==3.141.0
 urllib3==1.25.10
+spacy==2.3.2
```
Comment (damithc) [inline requirements.txt:5]: @joel-lim Do we still need this?

Comment (joel-lim) [reply requirements.txt:5]: My bad, I forgot to commit the new requirements.txt

**Comment by damithc** [comment]
Thanks @joel-lim

**Pull request merged**